phpBB/includes/ucp/ucp_pm_compose.php

```diff
@@ -828,7 +828,7 @@ function compose_pm($id, $mode, $action, $user_folders = array())
 
 		$enable_bbcode 		= (!$bbcode_status || isset($_POST['disable_bbcode'])) ? false : true;
 		$enable_smilies		= (!$smilies_status || isset($_POST['disable_smilies'])) ? false : true;
-		$enable_urls 		= (isset($_POST['disable_magic_url'])) ? 0 : 1;
+		$enable_urls 		= (!$url_status || isset($_POST['disable_magic_url'])) ? 0 : 1;
```

Comment:
Will it work like that?
```php
$enable_urls 		= ($url_status || !$request->is_set_post('disable_magic_url'));
```

Also, it probably requires `&&` as far as disallowing links in posts means disallowing them regardless of `'disable_magic_url'` option.

Reply:
No it needs to be an OR actually, exactly like the smilies and bbcode ones above